biojava-core/src/main/java/org/biojava/nbio/core/search/io/blast/BlastXMLParser.java

```diff
@@ -0,0 +1,232 @@
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.xml.sax.SAXException;
+
+/**
+ * Re-designed by Paolo Pavan on the footprint of: 
+ * org.biojava.nbio.genome.query.BlastXMLQuery by Scooter Willis <willishf at gmail dot com>
+ * 
+ * You may want to find my contacts on Github and LinkedIn for code info 
+ * or discuss major changes.
+ * https://github.com/paolopavan
+ * 
+ *
+ * @author Paolo Pavan
+ */
+public class BlastXMLParser implements ResultFactory {
+    private static final Logger log = Logger.getLogger(BlastXMLParser.class.getName());
+    Document blastDoc = null;
+    private File targetFile;
+    private List<Sequence> queryReferences, databaseReferences;
+    private Map<String,Sequence> queryReferencesMap, databaseReferencesMap;
+    
+    public BlastXMLParser() {
+        
+    }
+    public void setFile(File f){
+        targetFile = f;
+    }
+    
+    private void readFile(String blastFile) throws IOException, ParseException{
+        log.info("Start reading " + blastFile);
+        try {
+            blastDoc = XMLHelper.loadXML(blastFile);
+        } catch (SAXException ex) {
+            Logger.getLogger(BlastXMLParser.class.getName()).log(Level.SEVERE, null, ex);
+            throw new ParseException(ex.getMessage(),0);
+        } catch (ParserConfigurationException ex) {
+            Logger.getLogger(BlastXMLParser.class.getName()).log(Level.SEVERE, null, ex);
+            throw new ParseException(ex.getMessage(),0);
+        }
+        log.info("Read finished");
+    }
+    
+    public List<Result> createObjects(double maxEScore) throws IOException, ParseException {
+        if (targetFile == null) throw new IllegalStateException("File to be parsed not specified.");
+        
+        // getAbsolutePath throws SecurityException
+        readFile(targetFile.getAbsolutePath());
+        // create mappings between sequences and blast id
+        mapIds();
+        
+        ArrayList<Result> resultsCollection;
+        ArrayList<Hit> hitsCollection;
+        ArrayList<Hsp> hspsCollection;
+        
+        try {
+            // select top level elements
+            String program = XMLHelper.selectSingleElement(blastDoc.getDocumentElement(),"BlastOutput_program").getTextContent();
+            String version = XMLHelper.selectSingleElement(blastDoc.getDocumentElement(),"BlastOutput_version").getTextContent();
+            String reference = XMLHelper.selectSingleElement(blastDoc.getDocumentElement(),"BlastOutput_reference").getTextContent();
+            String dbFile = XMLHelper.selectSingleElement(blastDoc.getDocumentElement(),"BlastOutput_db").getTextContent();
+
+            log.info("Query for hits in "+ targetFile);
+            ArrayList<Element> IterationsList = XMLHelper.selectElements(blastDoc.getDocumentElement(), "BlastOutput_iterations/Iteration[Iteration_hits]");
+            log.info(IterationsList.size() + " results");
+
+            resultsCollection = new ArrayList();
```

Comment:
Dear Jose,
As I previously mentioned, type safe if guaranteed here since I previously have declared ArrayList<Result> resultsCollection. 
Sentences like resultsCollection.add(new String()) for example will already fail at compilation time. I  usually prefer it, because is less verbose and in my opinion cleaner. But if you really think it is important I can add it, just I can't do it until next week. 


> Il giorno 13 ott 2015, alle ore 19:12, Jose Manuel Duarte <notifications@github.com> ha scritto:
> 
> In biojava-core/src/main/java/org/biojava/nbio/core/search/io/blast/BlastXMLParser.java:
> 
> > +        ArrayList<Result> resultsCollection;
> > +        ArrayList<Hit> hitsCollection;
> > +        ArrayList<Hsp> hspsCollection;
> > +        
> > +        try {
> > +            // select top level elements
> > +            String program = XMLHelper.selectSingleElement(blastDoc.getDocumentElement(),"BlastOutput_program").getTextContent();
> > +            String version = XMLHelper.selectSingleElement(blastDoc.getDocumentElement(),"BlastOutput_version").getTextContent();
> > +            String reference = XMLHelper.selectSingleElement(blastDoc.getDocumentElement(),"BlastOutput_reference").getTextContent();
> > +            String dbFile = XMLHelper.selectSingleElement(blastDoc.getDocumentElement(),"BlastOutput_db").getTextContent();
> > +
> > +            log.info("Query for hits in "+ targetFile);
> > +            ArrayList<Element> IterationsList = XMLHelper.selectElements(blastDoc.getDocumentElement(), "BlastOutput_iterations/Iteration[Iteration_hits]");
> > +            log.info(IterationsList.size() + " results");
> > +
> > +            resultsCollection = new ArrayList();
> Missing type. There's still a few missing types for generics here and elsewhere in the pull request. Just follow any warnings that eclipse gives and try to solve as many as you can. If you can try to tidy those up that would be great. After that I think it would be ready for merging, nice work!
> 
> —
> Reply to this email directly or view it on GitHub.
>